lib/src/rules/avoid_redundant_argument_values.dart

```diff
@@ -79,7 +81,17 @@ class _Visitor extends SimpleAstVisitor {
         }
         final expressionValue = context.evaluateConstant(arg);
         if (expressionValue.value == value) {
-          rule.reportLint(arg);
+          var okToRemove = true;
```

Comment:
It isn't entirely clear to me which semantics you're trying to implement here, but I don't think this implements what you wanted. Given the TODO, I think you're trying to simply disable the lint for optional positional arguments. If that's the case, then I think you can just extend the condition on 83 with `&& (!param.isOptionalPositional || i == arguments.length - 1)`, and revert this change other than keeping the TODO.

Reply:
Thanks.  Updated as per our conversation.